### PR TITLE
feat(*): add Apollo server

### PR DESCRIPTION
**Pull request by burntcarrot**
This PR adds:

- Apollo core
- support for parsing configurations
- support for OpenAPI spec documentation through Swagger
- support for logging through Zap
- unit tests for Apollo

### CODE REVIEW

File: cmd/apollo/routes/routes.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package routes
+
+import (
+	"github.com/deepsourcelabs/apollo/controllers/dependency"
+	"github.com/deepsourcelabs/apollo/controllers/health"
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo/v4/middleware"
```
Comment (vishnu-deepsource) [inline cmd/apollo/routes/routes.go:7]: This dependency should not leak into cmd.  cmd should ideally be used only for service initialization.  Would recommend keeping it together with the handlers..

File: entity/health/health.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	return u.Repo.CreateService(ctx, domain, id)
+}
+
+func (u *Usecase) GetServices(ctx context.Context, id string) ([]Domain, error) {
+	ctx, cancel := context.WithTimeout(ctx, u.ctxTimeout)
+	defer cancel()
+
+	return u.Repo.GetServicesByID(ctx, id)
+}
+
```
Comment (vishnu-deepsource) [inline entity/health/health.go:33]: Entity should not have use cases.

File: entity/health/domain.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package health
+
+import "context"
+
+type Domain struct {
```
Comment (vishnu-deepsource) [inline entity/health/domain.go:5]: Add documentation.

Comment (vishnu-deepsource) [reply entity/health/domain.go:5]: What is a domain?

File: entity/health/health.go (hunk 2 above)
Comment (vishnu-deepsource) [reply entity/health/health.go:33]: Either keep all logic of a domain in one package, or split them up.

File: utils/config.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
+package utils
```
Comment (vishnu-deepsource) [inline utils/config.go:1]: Why is config kept in a utils directory?  Does not make sense to me.

File: utils/config.go [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+	File string `koanf:"file"`
+}
+
+type timeoutConfig struct {
+	Value int    `koanf:"value"`
+	Type  string `koanf:"type"`
+}
+
+// GetConfig returns a populated Config.
+func GetConfig(filePath string) (*Config, error) {
```
Comment (vishnu-deepsource) [inline utils/config.go:39]: Use viper or a better package.  This does not make sense.,

**Review by vishnu-deepsource — CHANGES_REQUESTED** [review]
Not sure why this was structured like this.  Also please send smaller PRs.  Would recommend reworking this from scratch.  

Please define all the domain models first and get that reviewed.

**Comment by burntcarrot** [comment]
Closing this. Will make changes through smaller PRs.

**Pull request closed without merging**